Update config.md

TehPeGaSuS opened the pull request:
Fixed:
- CONVOS_WEBIRC_PASSWORD_LOCALHOST not being displayed properly due to markdown malformation

Added:
- User systemd unit file
- Instructions how to enable user systemd without root rights and respective user commands

TehPeGaSuS committed:
Update config.md

Fixed:
- CONVOS_WEBIRC_PASSWORD_LOCALHOST not being displayed properly due to markdown malformation

Added:
- User systemd unit file
- Instructions how to enable user systemd without root rights and respective user commands

TehPeGaSuS committed:
Update config.md

Fixed:
- File name for logs on systemd user file

jhthorsen commented on doc/config.md:
```diff
@@ -187,9 +187,9 @@ You can enable the WEBIRC extension by setting an environment variable per
 
 1. You have a connection ID "irc-localhost". (Shown as "localhost" in the sidebar)
-2. Set the following environment variable to enable WEBIRC:
+2. Set the following environment variable to enable WEBIRC:  
```
Please remove the extra space.

jhthorsen commented on doc/config.md:
```diff
@@ -310,6 +310,43 @@ service:
+### As a normal user
+_**NOTE**: This was tested on Ubuntu 18.04 and newer but should work on all distros that have systemd support_
+
+Here is an example of a user systemd file, that can be placed in `.config/systemd/user/convos.service`.
```
Please add $HOME to the path. Same below.

jhthorsen commented on doc/config.md:
```diff
@@ -310,6 +310,43 @@ service:
+
+To enable user systemd units you need to perform some extra steps, as follow:
+
+    loginctl enable-linger
```
Does this have to be run as root? Either way, please add an extra line break.

jhthorsen replied:
"Does this have to be run as root?" has not been answered.

jhthorsen commented on doc/config.md:
```diff
@@ -310,6 +310,43 @@ service:
+    loginctl enable-linger
+Then you need to create the `.config/systemd/user` folder mannually or using the following command:  
+    
+    systemctl --user enable systemd-tmpfiles-clean.timer && systemctl --user disable systemd-tmpfiles-clean.timer
```
What is systemd-tmpfiles-clean? Either way, I think it's easier if you just tell the user how to create the directory manually.

jhthorsen replied:
I'm still curious about what systemd-tmpfiles-clean is, and I still think it's more confusing than helpful.

jhthorsen commented on doc/config.md:
```diff
@@ -310,6 +310,43 @@ service:
+    [Service]
+    # Replace any occurrence of $USER with your shell username
+    Environment=CONVOS_HOME=/home/$USER/convos
+    Environment=CONVOS_REVERSE_PROXY=1
```
Please remove CONVOS_REVERSE_PROXY. It can potentially be harmful to have this in the default config, when not running behind a reverse proxy.

jhthorsen commented on doc/config.md:
```diff
@@ -310,6 +310,43 @@ service:
+    
+    [Service]
+    # Replace any occurrence of $USER with your shell username
+    Environment=CONVOS_HOME=/home/$USER/convos
```
Why not use the default? I think that using the default will make it easier when reading other instructions such as https://convos.chat/doc/faq#where-does-convos-store-logs-settings-and-uploaded-files

If using the default, then maybe the CONVOS_LOG_FILE need to be adjusted as well, and maybe you need a PreExecStart command to create that directory. (Not sure about that though)

jhthorsen reviewed (changes requested):
Thanks a lot for the PR! A lot of great information here, but I do think some things needs adjustments.

For the future: It's much easier to accept a PR if it only has one change. (This PR contains bug fixes and new information)

TehPeGaSuS committed:
Update config.md

- Fixed all requested reviews, hopefully

TehPeGaSuS committed:
Update config.md

- Forgot to add $HOME to the location

jhthorsen commented on doc/config.md:
```diff
@@ -310,6 +310,43 @@ service:
+To enable user systemd units you need to perform some extra steps as the user, which are the following:
+
+    loginctl enable-linger
+Then you need to create the `$HOME/.config/systemd/user` folder mannually or using the following command:  
```
"mannually"

TehPeGaSuS committed:
Update config.md

- Add explicit commands to manually create the `$HOME/.config/systemd/user` folder
- Fixed typo

jhthorsen commented on doc/config.md:
```diff
@@ -310,6 +310,52 @@ service:
+    
+**Automatically with the following command**
+
+        systemctl --user enable systemd-tmpfiles-clean.timer && systemctl --user disable systemd-tmpfiles-clean.timer
```
Why is the indentation different here from the code below?

TehPeGaSuS committed:
Update config.md

- Fixed identation

TehPeGaSuS committed:
Update config.md

- Removed `Environment=CONVOS_REVERSE_PROXY=1`, since we should never assume that a reverse proxy is being used.
- The `mkdir -p ~/.config/systemd/user` is easier than havinf to type 3 commands.
- On systemd, Environment variables can use `~` as `$HOME`

This should be good for merging

TehPeGaSuS committed:
Update config.md

- The `%h` is no longer used
- We can't depend on global dependencies, so `After=network.target` is useless

TehPeGaSuS commented:
I need to redo some things. This is a complete mess!!!
I'll open a new PR
